jsonrpc/eth_endpoint.go

```diff
@@ -155,6 +155,41 @@ func (e *Eth) GetBlockByHash(hash types.Hash, fullTx bool) (interface{}, error)
 	return toBlock(block, fullTx), nil
 }
 
+// GetHeaderByNumber returns the requested canonical block header.
+// * When blockNr is -1 the chain head is returned.
+// * When blockNr is -2 the pending chain head is returned.
+func (e *Eth) GetHeaderByNumber(number BlockNumber) (interface{}, error) {
+	num, err := GetNumericBlockNumber(number, e.store)
+	if err != nil {
+		return nil, err
+	}
+
+	block, ok := e.store.GetBlockByNumber(num, true)
```

Comment:
We do not need block body (the transactions) here, so it should be better to call `e.store.GetHeaderByNumber()`.

Note that since it returns `*types.Header` instance, you would need to adapt the `filterExtra` function to accept (and possibly return) the `*types.Header`.